### ISSUE TITLE
Setting environment through -Dmicronaut.environments not working

### ISSUE THREAD

**Issue by haaah** [micronaut-projects/micronaut-core#8387]
### Expected Behavior

Micronaut should detect the entered environment and uses the corresponding application.yml.

### Actual Behaviour

`println("Env Active Names -> ${environment.activeNames}")`
Env Active Names -> []

### Steps To Reproduce

1. Create a new micronaut app with build=gradle and lang=groovy
2. entered the VM options: -Dmicronaut.environments=docker
<img width="1042" alt="image" src="https://user-images.githubusercontent.com/8482031/203031674-a9f2a44c-63a6-4e7d-a710-88a39d718b35.png">
3. Gradle run

### Environment Information

- MacOS Big Sur
- micronaut version 3.7.4
- JDK Azul Zulu 17.0.4

### Version

3.7.4

**Comment by haaah** [comment]
how ever using MICRONAUT_ENVIRONMENTS environment variable works

<img width="1042" alt="image" src="https://user-images.githubusercontent.com/8482031/203033131-a5b56739-2cf1-4d08-813a-b3b573775d4b.png">

```
@Inject
Environment environment
.
.
.
println("Env Active Names -> ${environment.activeNames}")
```
Env Active Names -> [docker]

**Comment by graemerocher** [comment]
I think this only passes the system properties to Gradle but not to the application /cc @melix

**Comment by melix** [comment]
That's correct, Gradle isolates system properties of the forked processes, so this behavior is expected. You can, however, make it pass explicitly by adding a snippet like this one:

```gradle
tasks.withType(JavaExec) {
    systemProperty("micronaut.environments", System.getProperty("micronaut.environments"))
}
```

**Comment by haaah** [comment]
coming from grails where we uses gradle to build our project. Putting the environment in the VM Options does work as expected

<img width="1043" alt="image" src="https://user-images.githubusercontent.com/8482031/203250407-6a5b5544-fa80-4622-b2bd-869235e2993b.png">

```
class BootStrap {

    def init = { servletContext ->
        docker{
            println("Env -> " + Environment.current.name)
        }
    }
}

```

Env -> docker

Does micronaut gradle build/run works differently from grails build/run?

**Comment by graemerocher** [comment]
I think it is because the Grails Gradle plugin is a bit more opinionated about this //cc @melix 

https://github.com/grails/grails-gradle-plugin/blob/e61ee3c10c0e58a273d41124db75f55924e8f106/src/main/groovy/org/grails/gradle/plugin/web/GrailsWebGradlePlugin.groovy#L53
